### PR TITLE
Run the 2to3 fixers

### PR DESCRIPTION
**Pull request by patrick96**
The command run was
```
2to3 -f all -f buffer -f idioms -f set_literal -n --no-diff -w *.py
```

And I manually changed

```
list([*list comprehension*])
```

to

```
[*list comprehension*]
```

The fixers mainly converted filters and maps to list comprehension and
wrapped dict.values and other dict functions in a list() call.

### CODE REVIEW

File: generators/cpp_client.py [hunk 1]
```diff
@@ -287,7 +287,7 @@ def build_collision_table():
     global namecount
     namecount = {}
 
-    for v in module.types.values():
+    for v in list(module.types.values()):
```
Comment (Lomadriel) [inline generators/cpp_client.py:290]: I think this cast is quite useless.

File: generators/cpp_client.py [hunk 2]
```diff
@@ -589,7 +589,7 @@ def get_expr_field_names(expr):
         prefix.append(('', '', self))
 
     all_fields = _c_helper_resolve_field_names (prefix)
-    resolved_fields_names = list(filter(lambda x: x in all_fields.keys(), unresolved_fields_names))
+    resolved_fields_names = [x for x in unresolved_fields_names if x in list(all_fields.keys())]
```
Comment (Lomadriel) [inline generators/cpp_client.py:592]:
```suggestion
    resolved_fields_names = [x for x in unresolved_fields_names if x in all_fields.keys()]
```

File: generators/cpp_client.py [hunk 3]
```diff
@@ -2756,7 +2756,7 @@ def _c_accessors_list(self, field):
                 (cookie_type, self.c_reply_name, base_func_name))
     f.write('.SH ERRORS\n')
     if hasattr(self, "doc") and self.doc:
-        for errtype, errtext in self.doc.errors.items():
+        for errtype, errtext in list(self.doc.errors.items()):
```
Comment (Lomadriel) [inline generators/cpp_client.py:2759]: If I'm not wrong this is wrong no?
This transform a dict view into a list.
```suggestion
        for errtype, errtext in self.doc.errors.items():
```

File: generators/cpp_client.py [hunk 4]
```diff
@@ -2774,7 +2774,7 @@ def _c_accessors_list(self, field):
         see = ['.BR %s (3)' % 'xcb-requests']
         if self.doc.example:
             see.append('.BR %s (3)' % 'xcb-examples')
-        for seename, seetype in self.doc.see.items():
+        for seename, seetype in list(self.doc.see.items()):
```
Comment (Lomadriel) [inline generators/cpp_client.py:2777]:
```suggestion
        for seename, seetype in self.doc.see.items():
```

File: generators/cpp_client.py [hunk 5]
```diff
@@ -2904,7 +2904,7 @@ def _c_complex_field(self, field, space=''):
         see = ['.BR %s (3)' % 'xcb_generic_event_t']
         if self.doc.example:
             see.append('.BR %s (3)' % 'xcb-examples')
-        for seename, seetype in self.doc.see.items():
+        for seename, seetype in list(self.doc.see.items()):
```
Comment (Lomadriel) [inline generators/cpp_client.py:2907]:
```suggestion
        for seename, seetype in self.doc.see.items():
```

File: generators/parameter.py [hunk 6]
```diff
@@ -209,7 +209,7 @@ def append_call_string(list):
 
         # end: for index, param in enumerate(self.parameter):
 
-        for k, v in lenfields.items():
+        for k, v in list(lenfields.items()):
```
Comment (Lomadriel) [inline generators/parameter.py:212]:
```suggestion
        for k, v in lenfields.items():
```

**Review by Lomadriel — COMMENTED** [review]
I'm not a python expert but I think some migrations are quite wrong or useless.
I don't know how python cast dict to list but I don't really like these casts

**Comment by patrick96** [comment]
The reason the fixer made all these casts is because in python2 functions like `.values()`, `.keys()`, `.items()` on dictionaries returned a copy of the dictionary's internal list. In python3 those functions return a view.

I'm not totally sure how the semantics differ or if it's even relevant in those cases. But those casts to list are necessary if we want to preserve the exact same semantics.

**Comment by Lomadriel** [comment]
> The reason the fixer made all these casts is because in python2 functions like `.values()`, `.keys()`, `.items()` on dictionaries returned a copy of the dictionary's internal list. In python3 those functions return a view.

O_o
Nice improvement in python3.

**Review by Lomadriel — APPROVED** [review]

**Comment by patrick96** [comment]
>  Nice improvement in python3.

I'm not sure if this was sarcastic or not ^^

**Pull request merged**